panther_hardware_interfaces/src/roboteq_driver.cpp

```diff
@@ -0,0 +1,345 @@
+// Copyright 2023 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <panther_hardware_interfaces/roboteq_driver.hpp>
+
+#include <cmath>
+#include <future>
+
+#include <panther_hardware_interfaces/utils.hpp>
+
+// All ids and sub ids were read directly from the eds file. Lely CANopen doesn't have the option to
+// parse them based on the ParameterName. Additionally between version v60 and v80 ParameterName
+// changed, for example: Cmd_ESTOP (old), Cmd_ESTOP Emergency Shutdown (new)
+// As parameter names changed, but ids stayed the same, it will be better to just use ids directly
+
+namespace panther_hardware_interfaces
+{
+
+RoboteqDriver::RoboteqDriver(
+  ev_exec_t * exec, lely::canopen::AsyncMaster & master, uint8_t id,
+  std::chrono::milliseconds sdo_operation_timeout)
+: lely::canopen::FiberDriver(exec, master, id),
```

Comment:
```suggestion
RoboteqDriver::RoboteqDriver(
  const std::shared_ptr<lely::ev::Executor> & exec,
  const std::shared_ptr<lely::canopen::AsyncMaster> & master, const std::uint8_t id,
  const std::chrono::milliseconds & sdo_operation_timeout)
: lely::canopen::FiberDriver(*exec, *master, id),
```

This is open suggestion on how this could be modified